### PR TITLE
Fixes #2480 Created separation for make file for Windows and Linux

### LINKED ISSUE #2480: [FEAT] Improve contribution guide
Related to #2455 - make the contribution guide better at helping new contributors. 
@hjain5164 you may be interested in this?

### PR DESCRIPTION
**Pull request by hjain5164**
#### Reference Issues/PRs

Fixes #2480 by adding the make file for Linux and windows separately so that native windows development of Deepchecks can be done by the developers.

#### What does this implement/fix? Explain your changes.
I have added a parent makefile which includes either the makefile.linux or makefile.windows based on the operating system.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
-   tests <#running-unit-tests>`__
+-  Refer to the `Windows users <#windows-users>`__ section if your development environment
+   is Windows.
+
+   - `Creating Virtual Environment <#creating-virtual-environment>`__
+   - `Installing Dependencies <#installing-dependencies>`__
+   - `Running Unit Tests <#windows-running-unit-tests>`__
+   - `Coding Style <#windows-coding-style>`__
+
+-  Refer to the `Linux/Mac users <#linux-and-mac-users>`__ section if your development environment
```
Comment (noamzbr) [inline CONTRIBUTING.rst:36]: Put this section above windows, as it's more common IMO.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
+-  Make sure that your code changes complies with the coding style of the
+   project.
 -  Keep API compatibility in mind when you change code. Reviewers of
    your pull request will comment on any API compatibility issues.
 
+Windows Users
+---------------
+For contributors using Windows operating system, you have to manually run
+the following commands since as of now the ``make`` commands works with
+Linux/Mac OS. **BTW, you are open to solve this issue!!**
```
Comment (noamzbr) [inline CONTRIBUTING.rst:63]:
```suggestion
Linux/Mac OS. **BTW, If you have a fix for that you are more than welcome to contribute!**
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
+   - Guard against future breaking changes to lower the maintenance cost.
+
 -  Bug fixes also generally require unit tests, because the presence of
    bugs usually indicates insufficient test coverage.
+-  Make sure that your code changes complies with the coding style of the
+   project.
 -  Keep API compatibility in mind when you change code. Reviewers of
    your pull request will comment on any API compatibility issues.
 
+Windows Users
```
Comment (noamzbr) [inline CONTRIBUTING.rst:59]: Move windows below the mac/linux part

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
+------------------------
+Installing dependencies
+------------------------
+.. _windows-installing-dependencies:
+
+Once your virtual environment is set up, activate your virtual environment
+by executing the command: ``./venv/Scripts/Activate.ps1``. Then, install
+the dependencies for your virtual environment by running the following command:
+
+- To install development dependencies:
```
Comment (noamzbr) [inline CONTRIBUTING.rst:84]: Forgot the regular dependencies(:

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
+
+- To install vision development dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\vision-requirements.txt
+
+..
+
+   If you are working with NLP module, then you may require to install
```
Comment (noamzbr) [inline CONTRIBUTING.rst:110]: fasttext doesn't just install from pip on windows? I know that on some machines it requires a specific gcc version or such...

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
 
+Windows Users
+---------------
+For contributors using Windows operating system, you have to manually run
+the following commands since as of now the ``make`` commands works with
+Linux/Mac OS. **BTW, you are open to solve this issue!!**
+
+
+----------------------------
+Creating virtual environment
```
Comment (noamzbr) [inline CONTRIBUTING.rst:67]: Please use the title standard that exists in other rst files. Usually the double ----- titles are "higher" than the single ----- titles.

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
+
+Once you have installed all the dependencies, you are ready to
+work on the project.
+
+-------------------
+Running Unit Tests
+-------------------
+.. _windows-running-unit-tests:
+
+To verify and execute all the unit tests, run the following command:
```
Comment (noamzbr) [inline CONTRIBUTING.rst:127]: all these run pylint, not tests.
You want to run pytest.

File: CONTRIBUTING.rst [hunk 8]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
+   # To run pylint on specific file
+   pylint .\deepchecks\nlp\utils\text_properties.py
+
+   # To run pydocstyle on all the files
+   python -m pydocstyle --convention=pep257 --add-ignore=D107 deepchecks
+
+
+Linux and Mac Users
+--------------------
+For contributors using either Linux or Mac operating system, we have included
```
Comment (noamzbr) [inline CONTRIBUTING.rst:168]:
```suggestion
For contributors using either Linux or Mac operating systems, we have created
```

File: CONTRIBUTING.rst [hunk 9]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
+a ``makefile`` which help you get all the setup done in just a few commands.
```
Comment (noamzbr) [inline CONTRIBUTING.rst:169]:
```suggestion
a ``makefile`` which will help you get all the setup done in just a few commands.
```

File: CONTRIBUTING.rst [hunk 10]
```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
...
+The ``make`` commands will create the virtual environment and install all the
```
Comment (noamzbr) [inline CONTRIBUTING.rst:170]: What will just running "make" do? Did you test that?

File: CONTRIBUTING.rst (hunk 7 above)
Comment (hjain5164) [reply CONTRIBUTING.rst:192]: Done

File: CONTRIBUTING.rst (hunk 6 above)
Comment (hjain5164) [reply CONTRIBUTING.rst:128]: Done

File: CONTRIBUTING.rst (hunk 5 above)
Comment (hjain5164) [reply CONTRIBUTING.rst:176]: Yeah actually on Windows, it is quite a difficult/unclear process. The best and easiest way is that if we have a wheel file, then we can directly install it on windows.

File: CONTRIBUTING.rst (hunk 4 above)
Comment (hjain5164) [reply CONTRIBUTING.rst:150]: Added

File: CONTRIBUTING.rst (hunk 1 above)
Comment (hjain5164) [reply CONTRIBUTING.rst:28]: Done

File: CONTRIBUTING.rst (hunk 10 above)
Comment (hjain5164) [reply CONTRIBUTING.rst]: No what I meant here was that all the make commands will have built-in commands to create the virtual env and no manual creation is required.

**Review by noamzbr — APPROVED** [review]

**Comment by noamzbr** [comment]
@all-contributors add @hjain5164 for docs

**Pull request merged**